pusher/src/Services/OpenIDClient.ts

```diff
@@ -26,7 +29,7 @@ class OpenIDClient {
     public authorizationUrl(state: string, nonce: string, playUri?: string, redirect?: string) {
         return this.initClient().then((client) => {
             return client.authorizationUrl({
-                scope: "openid email",
+                scope: "openid email " + OPID_ADDITIONAL_SCOPES,
```

Comment:
I think that we must define 'openid email' if `OPID_ADDITIONAL_SCOPES` doesn't existe and if `OPID_ADDITIONAL_SCOPES`exist, just use it and not concatenate `openid email` and `OPID_ADDITIONAL_SCOPES`?

Reply:
First of all the scope openid is required by specification. And from WA side, I think e-mail is also required. That was my idea behind additional scopes and adding them. We can change this of course, but it can lead to misconfiguration when one of the required scopes missing. So in the end your decision, I am able to properly set everything I need up on my side, but I just wanted to call attention to this.

Reply:
```
            let scopes : string[] = OPID_SCOPES.split(" ");
            if (scopes.indexOf("openid") == -1) {
                scopes.push("openid")
            }
            if (scopes.indexOf("email") == -1) {
                scopes.push("email")
            }
            return client.authorizationUrl({
                scope: scopes.join(" "),
```
I am a bit unsure if this is more elegant than before...

Reply:
You could use:

```js
if( OPID_ADDITIONAL_SCOPES. indexOf('email') === -1 || OPID_ADDITIONAL_SCOPES. indexOf('openid') === -1 ) {
  throw new Error('...');
}
```
```js
...
OPID_ADDITIONAL_SCOPES ? OPID_ADDITIONAL_SCOPES : "openid email"
...
```

Reply:
I okay, I thought you mean not throwing an error and instead adding them. Okay fine, I'll use that.